### ISSUE TITLE
Recursive text replacement

### ISSUE THREAD

**Issue by PavelGavlik** [madebysource/gulp-substituter#2]
It would great to replace text in files which we include.

Use case: I have multiple markup files for pages (index.html, about.html, ...) and I want common header.html and footer.html. But when I place some substituter tags in header.html, it won't work.

**Comment by dasuchin** [comment]
This feature would be awesome. Do you have a workaround? Also what method are you using to include a header.html and footer.html?

**Comment by PavelGavlik** [comment]
We're working on this feature, it should be available in few weeks. There is probably no workaround (except manually configuring gulp-substituter for partials and running it before Pho)

I have my substitute-config.js set up like this (note the .htm files without the l):

```
gulp = require('gulp');

module.exports = {
  analytics: 'UA-XXXXX-X',

  description: '',
  author: '',
  keywords: '',

  'og:url': '',
  'og:image': '',
  'og:type': '',
  'og:site_name': '',

  'twitter:title': '',
  'twitter:url': '',
  'twitter:image': '',
  'twitter:card': '',
  'twitter:site': '',
  'twitter:creator': '',

  header: function() { return gulp.src('src/partials/header.htm'); },
  footer: function() { return gulp.src('src/partials/footer.htm'); },
  angel_investing_nav: function() { return gulp.src('src/partials/angel-investing-nav.htm'); }
};
```

**Comment by PavelGavlik** [comment]
@jansedivy Any progress?

**Comment by PavelGavlik** [comment]
Implemented in 0.3.0

**Comment by jansedivy** [comment]
There is no extra configuration. It tries to replace any nested keys until there is nothing to replace
